ast/src/input/program_input/program_input.rs

```diff
@@ -23,6 +23,7 @@ use leo_input::{
 #[derive(Clone, PartialEq, Eq, Default)]
 pub struct ProgramInput {
     pub main: MainInput,
+    pub constants: ConstInput,
```

Comment:
I think rather than have an entirely different AST node that is identical we should minimize the number of different kind of nodes.

Option 1:
``` rust
pub struct ProgramInput {
    pub main: MainInput,
    pub constants: MainInput,
    registers: Registers,
}
```

Option 2:
``` rust
pub struct ProgramInput {
    pub inputs: Input,
    registers: Registers,
}

pub struct Input {
    // The tuple represents the input value and the constantness of the input.
    // Alternatively we can modify the InputValue struct to have the constantess. 
    input: IndexMap<String, Option<(InputValue, bool)>>,
}

```

Reply:
I literally had both designs 😊, but @collinc97 suggested I'd create absolutely different node. Yeah, both solutions look good to me, but in the end I decided to go with `ConstInput`.

Reply:
We should strive for a modular design. My recommendation was to separate the two scopes of variables in the `ProgramInput` struct. 

The struct should look like: 
```rust
pub struct ProgramInput {
    pub main: MainInput,
    pub constant: ConstantInput,
    registers:  Registers,
}
```

It should use the macro that I have already written [here](https://github.com/AleoHQ/leo/blob/master/ast/src/input/macros.rs)

Reply:
@collinc97 `MainInput` and `ConstInput` slightly differ from macro implementation. Do you think I should update both of them to only macro?

Reply:
Add them to the current macro or create a new one. I'll leave the decision to you. In both cases it resolves the comment from @gluax. We should not have duplicated code files.

Reply:
I agree. Thanks!

Reply:
Pushed new macro for both.